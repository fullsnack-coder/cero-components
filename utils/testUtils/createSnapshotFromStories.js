import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
