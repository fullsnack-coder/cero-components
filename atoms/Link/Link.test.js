import * as stories from './Link.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ atoms / Link ]', () => {
  describe('#Link', () => createSnapshotFromStories('Link', stories))
})
