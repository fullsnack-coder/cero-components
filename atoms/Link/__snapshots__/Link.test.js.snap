// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ atoms / Link ] #Link when 'Link' Colors is mounted should render 1`] = `
<DocumentFragment>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    class="link color-tertiary"
  >
    <p
      class="paragraph color-tertiary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</DocumentFragment>
`;

exports[`[ atoms / Link ] #Link when 'Link' Default is mounted should render 1`] = `
<DocumentFragment>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</DocumentFragment>
`;

exports[`[ atoms / Link ] #Link when 'Link' Sizes is mounted should render 1`] = `
<DocumentFragment>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-xs weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-sm weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-md weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
  <a
    class="link color-primary"
  >
    <p
      class="paragraph color-primary size-lg weight-semibold is-inline"
    >
      Mouths Muil
    </p>
  </a>
</DocumentFragment>
`;
